tests/net/lib/coap_client/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.0)
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(coap_client_test)
+
+set(APP_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+
+# Add test sources
+target_sources(app PRIVATE ${APP_SRC_DIR}/main.c)
+target_sources(app PRIVATE ${APP_SRC_DIR}/stubs.c)
+target_sources(app PRIVATE ${ZEPHYR_BASE}/subsys/net/lib/coap/coap_client.c)
+target_sources(app PRIVATE ${ZEPHYR_BASE}/subsys/net/lib/coap/coap.c)
+
+# Add includes directories
+target_include_directories(app PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src)
+target_include_directories(app PRIVATE ${ZEPHYR_BASE}/include/)
+
+add_compile_definitions(CONFIG_NET_SOCKETS_POLL_MAX=3)
+add_compile_definitions(CONFIG_COAP_ASYNC_CLIENT=y)
+add_compile_definitions(CONFIG_COAP_ASYNC_CLIENT_BUFFER_SIZE=256)
+add_compile_definitions(CONFIG_COAP_ASYNC_CLIENT_STACK_SIZE=1024)
+add_compile_definitions(CONFIG_COAP_LOG_LEVEL=4)
+add_compile_definitions(CONFIG_NET_SOCKETS_POSIX_NAMES=y)
+add_compile_definitions(CONFIG_COAP_INIT_ACK_TIMEOUT_MS=2000)
```

Comment:
Why aren't these set in `prj.conf`?

Reply:
To isolate the tests from other parts of code and omit some of the KConfig dependencies.